test/src/tests.js

```diff
@@ -139,4 +174,577 @@ describe('Bing Ads Event Forwarder', function() {
+                            },
+                        };
+                    },
+                },
+            };
+
+            mParticle.forwarder.process(obj);
+
+            // UETQ should now have 7 elements
+            // The first 3 elements of this array will be the initial consent payload
+            // The next 3 elments will be the consent update payload
+            // The 7th element will be the event payload
+
+            window.uetq.length.should.eql(7);
+            window.uetq[3].should.equal('consent');
+            window.uetq[4].should.equal('update');
+            window.uetq[5].should.eql(expectedUpdatedConsentPayload[2]);
+
+            done();
+        });
+
+        it('should NOT construct a Consent State Update Payload if consent DOES NOT change', function(done) {
+            mParticle.forwarder.init(
+                {
+                    tagId: 'tagId',
+                    consentMappingWeb: JSON.stringify(consentMap),
+                },
+                reportService.cb,
+                false // Disable testMode so we can test init
+            );
+
+            var expectedInitialConsentPayload = [
+                'consent',
+                'default',
+                { ad_storage: 'granted' },
+            ];
+
+            // UETQ queues up events as array elements and then parses them internally.
+            // The first 3 elements of this array will be the consent payload
+
+            window.uetq.length.should.eql(3);
+            window.uetq[0].should.equal('consent');
+            window.uetq[1].should.equal('default');
+            window.uetq[2].should.eql(expectedInitialConsentPayload[2]);
+
+            var obj = {
+                EventDataType: MessageType.PageEvent,
+                EventName: 'Test Page Event',
+                CustomFlags: {
+                    'Bing.EventValue': 10,
+                },
+                ConsentState: {
+                    getGDPRConsentState: function() {
+                        return {
+                            marketing_consent: {
+                                Consented: true,
+                                Timestamp: 1557935884509,
+                                ConsentDocument: 'Marketing_Consent',
+                                Location: 'This is fake',
+                                HardwareId: '123456',
+                            },
+                        };
+                    },
+
+                    getCCPAConsentState: function() {
+                        return {
+                            data_sale_opt_out: {
+                                Consented: false,
+                                Timestamp: Date.now(),
+                                Document: 'some_consent',
+                            },
+                        };
+                    },
+                },
+            };
+
+            mParticle.forwarder.process(obj);
+
+            // UETQ should now have 7 elements
```

Comment:
```suggestion
            // UETQ should now have 4 elements
```

Maybe another good test would be like window.uetq.indexOf('update') === -1

Since the idea is there is no update to consent.